### PR TITLE
BUG: Fix Ability to set both color and style in pandas plotting

### LINKED ISSUE #59461: BUG: Ability to set both color and style in pandas plotting
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
df = pd.DataFrame({
'pig': [20, 18, 489, 675, 1776],
 'horse': [4, 25, 281, 600, 1900]
}, index=[1990, 1997, 2003, 2009, 2014])

color={'pig':'black','horse':'brown'}
style={'pig':'solid','horse':'dashed'}
df.plot.line(color=color,style=style)

plt.show()
```

### Issue Description

In above code we can pass color or style but not both.

### Expected Behavior

We should be able to change both color and style.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit                : d9cdd2ee5a58015ef6f4d15c7226110c9aab8140
python                : 3.9.6.final.0
python-bits           : 64
OS                    : Darwin
OS-release            : 23.5.0
Version               : Darwin Kernel Version 23.5.0: Wed May  1 20:19:05 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T8112
machine               : arm64
processor             : arm
byteorder             : little
LC_ALL                : None
LANG                  : en_CA.UTF-8
LOCALE                : en_CA.UTF-8

pandas                : 2.2.2
numpy                 : 1.26.3
pytz                  : 2023.3.post1
dateutil              : 2.8.2
setuptools            : 58.0.4
pip                   : 21.2.4
Cython                : None
pytest                : 8.2.2
hypothesis            : None
sphinx                : None
blosc                 : None
feather               : None
xlsxwriter            : None
lxml.etree            : 5.1.1
html5lib              : 1.1
pymysql               : None
psycopg2              : None
jinja2                : 3.1.2
IPython               : 8.18.1
pandas_datareader     : None
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : 4.12.2
bottleneck            : None
dataframe-api-compat  : None
fastparquet           : None
fsspec                : None
gcsfs                 : None
matplotlib            : 3.8.2
numba                 : None
numexpr               : None
odfpy                 : None
openpyxl              : None
pandas_gbq            : None
pyarrow               : 15.0.0
pyreadstat            : None
python-calamine       : None
pyxlsb                : None
s3fs                  : None
scipy                 : 1.12.0
sqlalchemy            : None
tables                : None
tabulate              : 0.9.0
xarray                : None
xlrd                  : None
zstandard             : None
tzdata                : 2023.4
qtpy                  : 2.4.1
pyqt5                 : None

</details>

### PR DESCRIPTION
**Pull request by AntiKnot**
- [x] closes #59461  (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/plotting/_matplotlib/core.py [hunk 1]
```diff
@@ -455,6 +455,8 @@ def _validate_color_args(self, color, colormap):
                 styles = self.style
             else:
                 styles = [self.style]
+            if isinstance(self.style, dict):
```
Comment (mroeschke) [inline pandas/plotting/_matplotlib/core.py:458]: Can you make this an `elif` incorporating it in the above if/else block?

File: doc/source/whatsnew/v3.0.0.rst [hunk 2]
```diff
@@ -634,6 +634,7 @@ Plotting
 - Bug in :meth:`.DataFrameGroupBy.boxplot` failed when there were multiple groupings (:issue:`14701`)
 - Bug in :meth:`DataFrame.plot` that causes a shift to the right when the frequency multiplier is greater than one. (:issue:`57587`)
 - Bug in :meth:`Series.plot` with ``kind="pie"`` with :class:`ArrowDtype` (:issue:`59192`)
+- Bug in :meth:`core._validate_color_args` raising ``ValueError`` when set both color and style in plt.plot.line (:issue:`59461`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v3.0.0.rst:637]:
```suggestion
- Bug in :meth:`DataFrame.plot.line` raising ``ValueError`` when set both color and a ``dict`` style (:issue:`59461`)
```

File: pandas/plotting/_matplotlib/core.py (hunk 1 above)
Comment (AntiKnot) [reply pandas/plotting/_matplotlib/core.py]: Refactoring conditional statements block.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @AntiKnot

**Pull request merged**